### PR TITLE
Dirty fix for App hanging when windows are invisible on WindowsOS

### LINKED ISSUE #14135: visible: false on WindowPlugin Stops Rendering
## Bevy version
0.14

## Relevant system information
```ignore
`SystemInfo { os: "Windows 11 Home", kernel: "22635", cpu: "AMD Ryzen 7 5800X3D 8-Core Processor", core_count: "8", memory: "31.9 GiB" }`
`AdapterInfo { name: "NVIDIA GeForce RTX 4060 Ti", vendor: 4318, device: 10243, device_type: DiscreteGpu, driver: "", driver_info: "", backend: Dx12 }`
```

## What you did

Set visible: false on WindowPlugin

## What went wrong

Systems added to "Update" no longer trigger.

Using:
```
fn setup_window(mut window: Query<&mut Window>, frames: Res<FrameCount>) {
    println!("{}", frames.0);
    if frames.0 == 3 {
        window.single_mut().visible = true;
    }
}
```

If visible is set to false by default, we only see `0` in the console, once. If set to true, we see the frame count as expected for every frame.

## Additional information
Also happens on Bevy example found [here](https://bevyengine.org/examples-webgpu/window/window-settings/)

### PR DESCRIPTION
**Pull request by MiniaczQ**
# Objective

- Fixes #14135 

## Solution

- If no windows are visible, app updates will run regardless of redraw call result.

This a relatively dirty fix, a more robust solution is desired in the long run:
https://github.com/bevyengine/bevy/issues/1343#issuecomment-770091684
https://discord.com/channels/691052431525675048/1253771396832821270/1258805997011730472
The solution would disconnect rendering from app updates.

## Testing

- `window_settings` now works

## Other platforms

Not a problem on Linux: https://discord.com/channels/691052431525675048/692572690833473578/1259526650622640160
Not a problem on MacOS: https://discord.com/channels/691052431525675048/692572690833473578/1259563986148659272

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Can you create an issue for the proper fix and link it here please?

File: crates/bevy_winit/src/state.rs [hunk 1]
```diff
@@ -505,7 +507,9 @@ impl<T: Event> ApplicationHandler<T> for WinitAppRunnerState<T> {
 
         if should_update {
             // Not redrawing, but the timeout elapsed.
-            if !self.ran_update_since_last_redraw {
+            // NOTE: Additional condition for WindowsOS.
```
Comment (alice-i-cecile) [inline crates/bevy_winit/src/state.rs:510]: This should link the issue for the "proper fix".

Comment (alice-i-cecile) [inline crates/bevy_winit/src/state.rs:510]: Are we certain that this is isolated to Windows?

Comment (MiniaczQ) [reply crates/bevy_winit/src/state.rs:510]: I've asked folk on discord to help test this on Linux and MacOS, I'll test it on WASM

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
This is a vital fix for 0.14.1, because of the impact on servers in networked games.

**Comment by MiniaczQ** [comment]
> This is a vital fix for 0.14.1, because of the impact on servers in networked games.

Actually, a headless server works just fine, it's only when we try to use a window driven event loop with invisible windows

**Review by mockersf — APPROVED** [review]

**Pull request merged**